Source/WebGPU/WebGPU/RenderPassEncoder.h

```diff
@@ -105,6 +105,7 @@ class RenderPassEncoder : public WGPURenderPassEncoderImpl, public RefCounted<Re
     const Ref<Device> m_device;
     MTLPrimitiveType m_primitiveType { MTLPrimitiveTypeTriangle };
     id<MTLBuffer> m_indexBuffer { nil };
+    id<MTLRenderPipelineState> m_renderPipeline { nil };
```

Comment:
The validation in the specification seems to be a bit complicated: https://www.w3.org/TR/webgpu/#abstract-opdef-validate-encoder-bind-groups

We need to store all the bind groups which are set on the encoder somewhere and ensure they are ["group equivalent"](https://www.w3.org/TR/webgpu/#group-equivalent) with the current pipeline.

There is no way afaik to do that as setBindGroup is called, since setPipeline could be called multiple times on the same encoder.

I filed https://bugs.webkit.org/show_bug.cgi?id=254307 to track that and updated this PR to only store a boolean as suggested.